setup.py

```diff
@@ -16,90 +16,6 @@
 # limitations under the License.
 #
 
-from setuptools import setup, find_packages
-import os
-import re
+from setuptools import setup
 
-
-def parse_requirements():
-    '''
-    Parses the requirements.txt file to extras install and extra requirements.
-    Sections headed with # [extra] are assigned to the 'extra' extra.
-
-    Returns:
-        Tuple with install and extra requires passible to :func:`setuptools.setup`.
-    '''
-    with open('requirements.txt', 'rt') as f:
-        lines = f.readlines()
-
-    extras_require = {}
-    requires = []
-    all_requires = []
-    current = None
-    for line in lines:
-        line = line.strip()
-
-        if line == '':
-            continue
-
-        match = re.match(r'^#\s*\[([a-zA-Z0-9_]+)\]$', line)
-        if match:
-            extra = match.group(1)
-            current = list()
-            extras_require[extra] = current
-        elif line.startswith('#'):
-            continue
-        else:
-            line = line.split('#')[0].strip()
-            if current is None:
-                requires.append(line)
-            else:
-                current.append(line)
-                all_requires.append(line)
-
-    extras_require['all'] = all_requires
-
-    return requires, extras_require
-
-def main():
-    try:
-        for nexus_definition_dir in ("/./", "/base_classes/", "/applications/", "/contributed_definitions/"):
-            f = open(os.path.abspath(os.path.dirname(__file__)) + "/nexusparser/definitions/" + nexus_definition_dir + "__init__.py", "a")
-            f.close()
-    except FileNotFoundError:
-        pass
-
-    with open("README.md", "r") as f:
-        long_description = f.read()
-
-    requires, extras_require = parse_requirements()
-
-    setup(
-        name='nexusparser',
-        version='0.0.1',
-        description='NOMAD parser implementation for Nexus.',
-        long_description=long_description,
-        long_description_content_type="text/markdown",
-        author='The NOMAD Authors',
-        license='APACHE 2.0',
-        packages=find_packages(exclude=['tests']),
-        package_data={
-            'nexusparser.definitions.base_classes': ['*.xml'],
-            'nexusparser.definitions.applications': ['*.xml'],
-            'nexusparser.definitions.contributed_definitions': ['*.xml'],
-            'nexusparser.definitions': ['*.xsd']
-        },
-        include_package_data=True,
-        install_requires=requires,
-        extras_require=extras_require,
-        entry_points={
-            'console_scripts': [
-                'read_nexus = nexusparser.tools.nexus:main',
-                'dataconverter = nexusparser.tools.dataconverter.convert:convert_cli',
-                'yaml2nxdl = nexusparser.tools.yaml2nxdl.yaml2nxdl:launch_tool'
-            ]
-        })
-
-
-if __name__ == '__main__':
-    main()
+setup()
```

Comment:
will it ensure packaging the xml files from nexus definitinos?

Reply:
This is still in draft. There are many other developments happening in Nomad that I am waiting for. But I'll add this point as a TODO above so that we don't forget this before merging.